File: sdks/python/applad/deploy.py
"""Deploy service.

Deploy targets are the deployable unit. The API mounts them under
/deploy/targets (there is no flat /deploy resource); triggering a deploy runs
the target as an execution.
"""


class Deploy:
    def __init__(self, client):
        self.client = client

    def create(self, name: str, deploy_type: str, config: dict = None):
        body = {"name": name, "type": deploy_type}
        if config:
            body.update(config)
        return self.client._call("POST", "/deploy/targets", body)

    def list(self):
        return self.client._call("GET", "/deploy/targets")

    def get(self, target_id: str):
        return self.client._call("GET", f"/deploy/targets/{target_id}")

    def update(self, target_id: str, data: dict):
        return self.client._call("PUT", f"/deploy/targets/{target_id}", data)

    def delete(self, target_id: str):
        return self.client._call("DELETE", f"/deploy/targets/{target_id}")

    def deploy(self, target_id: str, options: dict = None):
        """Trigger a deploy of the target. Returns the created execution."""
        return self.client._call(
            "POST", f"/deploy/targets/{target_id}/executions", options or {}
        )

File: sdks/dart/lib/src/server/deploy.dart
import '../server_client.dart';

/// Deploy service (server-side) — manage deploy targets.
///
/// Deploy targets are the deployable unit. The API mounts them under
/// /deploy/targets (there is no flat /deploy resource); triggering a deploy
/// runs the target as an execution.
class Deploy {
  final ApplAdServer _client;

  Deploy(this._client);

  Future<Map<String, dynamic>> create({
    required String name,
    required String type,
    Map<String, dynamic>? config,
  }) async {
    return _client.post('/v1/deploy/targets', data: {
      'name': name,
      'type': type,
      if (config != null) ...config,
    });
  }

  Future<Map<String, dynamic>> list() async {
    return _client.get('/v1/deploy/targets');
  }

  Future<Map<String, dynamic>> get(String targetId) async {
    return _client.get('/v1/deploy/targets/$targetId');
  }

  Future<Map<String, dynamic>> update(
      String targetId, Map<String, dynamic> data) async {
    return _client.put('/v1/deploy/targets/$targetId', data: data);
  }

  Future<Map<String, dynamic>> delete(String targetId) async {
    return _client.delete('/v1/deploy/targets/$targetId');
  }

  /// Trigger a deploy of the target. Returns the created execution.
  Future<Map<String, dynamic>> deploy(String targetId,
      {Map<String, dynamic>? options}) async {
    return _client.post('/v1/deploy/targets/$targetId/executions',
        data: options ?? {});
  }
}
